app/assets/javascripts/admin/views/broken-links-report.js

```diff
@@ -0,0 +1,76 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function BrokenLinksReport (module) {
+    this.module = module
+  }
+
+  BrokenLinksReport.prototype.init = function () {
+    if (!fetch) return
+    var button = this.module.querySelector('button')
+    var authenticityTokenField = this.module.querySelector('[name="authenticity_token"]')
+    if (button && authenticityTokenField) this.setupEventListener(button, authenticityTokenField.value)
+    var refreshLink = this.module.querySelector('.js-broken-links-refresh')
+    if (refreshLink) {
+      this.setupPolling(refreshLink)
+    }
+  }
+
+  BrokenLinksReport.prototype.setupEventListener = function (button, authenticityToken) {
+    var action = this.module.querySelector('.js-broken-links-form').action
+    button.addEventListener('click', function (event) {
+      event.preventDefault()
+      fetch(action + '.json', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ authenticity_token: authenticityToken })
+      }).then(function (response) { return response.json() })
+        .catch(function () { location.reload() })
+        .then(function (json) {
+          this.replaceContentsAndCopyClasses(json.html)
+          this.init()
+        }.bind(this))
+    }.bind(this))
+  }
+
+  BrokenLinksReport.prototype.setupPolling = function (refreshLink) {
+    refreshLink.hidden = true
+
+    var retries = 10
+    var retry = function () {
+      retries -= 1
+      if (retries === 0) {
+        refreshLink.hidden = false
```

Comment:
if the refresh link shows, will the context change to let the user know that they need to try again?

Reply:
What kind of context change would you suggest?
The important thing for the user is seeing the report which involves navigating to this part of the page either way.
It's only viewing the report that requires either waiting or clicking the "Check progress" link.